### PR TITLE
New: Column config column items

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
We recently added `navi-column-config` behind a feature flag, letting you see and configure your selected columns

## Proposed Changes
- Add the columns to the column config

## Screenshots

| Demo | Screenshot |
|------|------------|
|      |            |
 

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/navi-column-config.js [hunk 1]
```diff
@@ -6,18 +6,114 @@
...
+    }
+
+    yield Promise.all([
+      ...keptSprites.map(sprite => move(sprite, { duration: moveDuration })),
+      ...insertedSprites.map(sprite => fadeIn(sprite, { duration: 100 })),
+      ...removedSprites.map(sprite => fadeOut(sprite, { duration: 0 }))
+    ]);
+  }
+
+  @computed('report.request.{metrics.[],dimensions.[],logicalTable.timeGrain}')
```
Comment (mattgraham1995) [inline packages/reports/addon/components/navi-column-config.js]: needs jsdoc

File: packages/reports/app/styles/navi-reports/components/navi-column-config.less [hunk 2]
```diff
@@ -3,21 +3,65 @@
...
+  &:hover {
+    opacity: 1;
+  }
+}
+
 .navi-column-config {
   display: flex;
-  overflow: hidden;
+  font-family: @font-family-sans-serif;
+  font-size: 14px;
```
Comment (mattgraham1995) [inline packages/reports/app/styles/navi-reports/components/navi-column-config.less]: variable

File: packages/reports/app/styles/navi-reports/components/navi-column-config.less [hunk 3]
```diff
@@ -3,21 +3,65 @@
...
+    border-bottom: 1px solid @denali-gray-300;
+
+    &__name {
+      flex: 3;
+    }
+
+    &__dropdown-icon {
+      .navi-column-config-icon;
+      flex: 0.5;
+      font-size: 12px;
```
Comment (mattgraham1995) [inline packages/reports/app/styles/navi-reports/components/navi-column-config.less]: variable

File: packages/reports/app/styles/navi-reports/components/navi-column-config.less [hunk 4]
```diff
@@ -3,21 +3,65 @@
...
+    }
+
+    &__remove-icon {
+      .navi-column-config-icon;
+      flex: 0.5;
+      text-align: center;
+      font-size: 16px;
```
Comment (mattgraham1995) [inline packages/reports/app/styles/navi-reports/components/navi-column-config.less]: variable

File: packages/reports/tests/acceptance/navi-report/navi-column-config-test.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    assert.deepEqual(getColumns(), ['Date Time (Day)'], 'Initially only the date time is visible');
+  });
+
+  test('time grain - switching and removing', async function(assert) {
+    assert.expect(5);
+    await visit('reports/1/view');
+
+    assert.dom('.filter-builder__subject').hasText('Date Time (Day)', 'Time grain is initially day');
+
+    await click(findAll('.navi-column-config-item__remove-icon')[0]); // Remove Date Time (Day)
```
Comment (mattgraham1995) [inline packages/reports/tests/acceptance/navi-report/navi-column-config-test.js]:
```suggestion
    await click('.navi-column-config-item__remove-icon'); // Remove Date Time (Day)
```

File: packages/reports/addon/components/navi-column-config.js [hunk 6]
```diff
@@ -6,18 +6,114 @@
...
+  /**
+   * @method getDisplayName
+   * @param {Object} asset
+   * @param {String} type
+   * @param {Object} visualization
+   * @returns {String} display name from visualization metadata or default display name for metric, dimension, or Date
+   */
+  getDisplayName(asset, type) {
+    const nameServiceMap = {
+      metric: metric => this.metricName.getDisplayName(metric.serialize()),
```
Comment (cythrawll) [inline packages/reports/addon/components/navi-column-config.js]: Can we add a TODO here to add namespace parameter when the service supports it? My next PR will add support.

File: packages/reports/addon/templates/components/report-view.hbs [hunk 7]
```diff
@@ -14,13 +18,13 @@
           @onVisualizationTypeUpdate={{this.onVisualizationTypeUpdate}}
       />
     </div>
-      {{#animated-if (and this.hasRequestRun (not this.isEditingVisualization)) use=this.visFadeTransition}}
+      {{#animated-if (and this.hasRequestRun (not this.isEditingVisualization)) use=this.visFadeTransition duration=200}}
```
Comment (jkusa) [inline packages/reports/addon/templates/components/report-view.hbs]: `visFadeTransition` has a hard coded duration on `insertedSprites`:

https://github.com/yahoo/navi/blob/master/packages/reports/addon/components/report-view.js#L194

File: packages/reports/app/styles/navi-reports/components/navi-column-config.less [hunk 8]
```diff
@@ -3,21 +3,65 @@
...
+    align-items: center;
+    display: flex;
+    cursor: pointer;
+    margin-left: 15px;
+    padding: 10px 0;
+    min-height: 20px;
+    border-bottom: 1px solid @denali-gray-300;
+
+    &__name {
+      flex: 3;
```
Comment (jkusa) [inline packages/reports/app/styles/navi-reports/components/navi-column-config.less]: Are we handling text overflow?

**Comment by kevinhinterlong** [comment]
> Are we handling text overflow?

Added, this is what it looks like now

column name overflow

**Pull request merged**